src/main/java/school/hei/haapi/service/utils/PdfRenderer.java

```diff
@@ -0,0 +1,33 @@
+package school.hei.haapi.service.utils;
+
+import static school.hei.haapi.model.exception.ApiException.ExceptionType.SERVER_EXCEPTION;
+
+import com.lowagie.text.DocumentException;
+import java.io.ByteArrayOutputStream;
+import java.util.function.Function;
+import org.springframework.stereotype.Component;
+import org.xhtmlrenderer.pdf.ITextRenderer;
+import school.hei.haapi.model.exception.ApiException;
+
+@Component
+public class PdfRenderer implements Function<String, byte[]> {
+  @Override
+  public byte[] apply(String html) {
+    //given
+    ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+    ITextRenderer renderer = new ITextRenderer();
+    renderer.setDocumentFromString(html);
+    renderer.layout();
+    
+    //when
+    try {
+      renderer.createPDF(outputStream);
+    }
+
+    //then
+    catch (DocumentException e) {
+      throw new ApiException(SERVER_EXCEPTION, e.getMessage());
+    }
+    return outputStream.toByteArray();
```

Comment:
remove the //given //when //then comments,

and merge the when and then part/

The given part is only the instanciation of variables and the setter,

The layout is part of the when